docs/datastore/cloudsql_mysql.md

```diff
@@ -0,0 +1,182 @@
+# Setup and configure Cloud SQL for MySQL
+
+## Before you begin
+
+1. Make sure you have a Google Cloud project and billing is enabled.
+
+1. Set your `PROJECT_ID` environment variable:
+
+    ```bash
+    export PROJECT_ID=<YOUR_PROJECT_ID>
+    ```
+
+1. [Install](https://cloud.google.com/sdk/docs/install) the gcloud CLI.
+
+1. Set gcloud project:
+
+    ```bash
+    gcloud config set project $PROJECT_ID
+    ```
+
+1. Enable APIs:
+
+    ```bash
+    gcloud services enable sqladmin.googleapis.com \
+                           aiplatform.googleapis.com
+    ```
+
+1. [Install python][install-python] and set up a python [virtual environment][venv].
+
+1. Make sure you have python version 3.11+ installed.
+
+    ```bash
+    python -V
+    ```
+
+1. Download and install [mysql-client cli (`mysql`)][install-mysql].
+
+1. Install the [Cloud SQL Auth Proxy client][install-cloudsql-proxy].
+
+[install-python]: https://cloud.google.com/python/docs/setup#installing_python
+[venv]: https://cloud.google.com/python/docs/setup#installing_and_using_virtualenv
+[install-mysql]: https://dev.mysql.com/doc/mysql-installation-excerpt/8.0/en/
+[install-cloudsql-proxy]: https://cloud.google.com/sql/docs/mysql/connect-auth-proxy
+
+
+## Create a Cloud SQL for MySQL instance
+
+1. Set environment variables. For security reasons, use a different password for
+   `$DB_PASS` and note it for future use:
+
+    ```bash
+    export DB_PASS=my-cloudsql-pass
+    export DB_USER=mysql
+    export INSTANCE=my-cloudsql-instance
+    export REGION=us-central1
+    ```
+
+1. Create a MySQL instance with vector enabled:
+
+    ```bash
+    gcloud sql instances create $INSTANCE \
+        --database-version=MYSQL_8_0_36 \
+        --cpu=4 \
+        --memory=16GB \
+        --region=$REGION \
+        --database-flags=cloudsql_vector=ON
+    ```
+
+1. Set password for mysql user:
+
+    ```bash
+    gcloud sql users set-password $DB_USER \
+        --instance=$INSTANCE \
+        --password=$DB_PASS
+    ```
+
+
+## Connect to the Cloud SQL instance
+
+1. Connect to instance using cloud sql proxy:
+
+    ```bash
+    ./cloud-sql-proxy $PROJECT_ID:$REGION:$INSTANCE
+    ```
+
+1. Verify you can connect to your instance with the `mysql` tool. Enter
+   password for Cloud SQL (`$DB_PASS` environment variable set above) when prompted:
+
+    ```bash
+    mysql "host=127.0.0.1 port=5432 sslmode=disable user=$DB_USER"
+    ```
+
+## Update config
+
+Update `config.yml` with your database information.
+
+```bash
+host: 0.0.0.0
+datastore:
+    # Example for cloudsql_mysql.py provider
+    kind: "cloudsql-mysql"
+    # Update this with your project ID
+    project: <PROJECT_ID>
+    region: us-central1
+    instance: my-cloudsql-instance
+    # Update this with the database name
+    database: "assistantdemo"
+    # Update with database user, the default is `mysql`
+    user: "mysql"
```

Comment:
If we decide to use `root` as the default user we need to update here
```suggestion
    # Update with database user, the default is `root`
    user: "root"
```

Reply:
done